src/escalation/jira_escalation.py

```diff
@@ -0,0 +1,226 @@
+    """ """
+
+    def __init__(
+        self,
+        jira: Jira,
+        slack_client: SlackClient,
+        slack_channel: str,
+        default_labels: list[str],
+        additional_labels: list[str],
+        default_jira_project: str,
+    ):
+        self.default_labels = default_labels
+        self.additional_labels = additional_labels
+        self.default_project = default_jira_project
+        self.slack_client = slack_client
+        self.jira = jira
+
+        # build query and process ocp lp firewatch tickets in NOACK and ACK status
+        jira_query = ""
+        for status in ["NO ACK", "ACK"]:
+            if default_jira_project:
+                jira_query = self.build_jql(status)
+            jira_query = self.build_jql(status)
+
+            LOGGER.info(f" jira-query: {jira_query}")
+
+            self.process_issues(jira_query, slack_channel)
+
+        # process PQE tickets
+        pqe_jira_query = (
+            f"Project != {default_jira_project} AND status not in('Resolved','Blocked','Closed','Backlog','Done')"
+        )
+
+        if self.additional_labels:
+            formatted_additional_labels = '","'.join(self.additional_labels)
+            LOGGER.info(f" add labels {formatted_additional_labels}")
+            pqe_jira_query += f' AND (labels IN("{formatted_additional_labels}"))'
+
+        if self.default_labels:
+            default_condition = " AND".join([f'labels = "{label}"' for label in self.default_labels])
+            pqe_jira_query += f" AND ({default_condition})"
+
+        LOGGER.info(f" jira query PQE tickets: {pqe_jira_query}")
+        self.process_issues(pqe_jira_query, slack_channel)
+
+    def process_issues(self, jira_query: str, slack_channel: str) -> None:
+        """ 
+        Process jira issues for escalation based on escaltion criteria
+        Args:
+          jira_query (string): JQL query string to get the issues
+          slack_channel (str): slack channel name to send notifications
+        """
+
+        issues_list = self.jira.search_issues(jql_query=jira_query)
+        LOGGER.info(f" Jira issues: {issues_list}")
+        # LOGGER.info(f"-------------------------Jira Issues----------------------------")
+        test_email = "vshakya@redhat.com"
+        for issue in issues_list:
+            jira_issue = self.jira.get_issue_by_id_or_key_with_changelog(issue)
+
+
+
+            LOGGER.info(f" ---------------jira issue: {jira_issue.key}--- Status : {jira_issue.fields.status.name}  ---------------")
+
+            if jira_issue.fields.assignee is None:
+                LOGGER.info(
+                    f" No assignee found for this jira issue : {jira_issue.key}"
+                )  # to do : send notification in slack to mpiit team for followup
+                issue_url = f'<https://issues.stage.redhat.com/browse/{jira_issue.key}|{jira_issue.key}>'
+                message = f"No assignee found for this jira issue :  {issue_url} \n please provide an update"
```

Comment:
Maybe add the related job name and team name, it will easier to follow up

Reply:
added the job name and url